### PR TITLE
feature/add-flow-brake-tests

### PR DESCRIPTION
**Pull request by sotnikov-s**
https://lido.myjetbrains.com/youtrack/issue/LSC-216

- implement messages to ignore support for `dc4bc_cli refresh_state` command + tests
- add tests for the following cases:
    - a participant edits a message that has been proposed to sign
    - a participant sends an invalid partial signature
    - a participant uses key from another DKG to sign a message
- several minor fixes/improvements

### CODE REVIEW

**Review by swelf19 — APPROVED** [review]
lgtm
but one thing to keep in mind, we barely know who is malicious participant and unlike flowtests we will have to ignore a random signature until the reconstruction is succeed

**Comment by sotnikov-s** [comment]
> lgtm but one thing to keep in mind, we barely know who is malicious participant and unlike flowtests we will have to ignore a random signature until the reconstruction is succeed

there seems to be no easy way to help people user-friendly recover after a participant tried to deceive the others. as I understand, we need the flow to be handy for proper usage and unable to be hacked -- these have been accomplished and tested.
if it's needed to check why signature has not been reconstructed, it's possible to explore kafka messages for a junk signature. if all messages are valid signatures, but one has been created using another dkg, we can't be sure which signature exactly is bad, can we?

File: client/flow_test.go [hunk 1]
```diff
@@ -716,41 +906,19 @@ func TestResetStateFlow(t *testing.T) {
...
-
-	runCtx, runCancel := context.WithCancel(context.Background())
-	for _, n := range nodes {
-		go n.run(nil, runCtx)
+	log.Print("\n\n\nResetting nodes states\n\n\n")
+	if err := resetNodesStates(nodes, []string{msgToIgnore}, false); err != nil {
+		t.Fatalf("failed to reset nodes states: %v", err)
 	}
-
+	time.Sleep(10 * time.Second)
```
Comment (oopcode) [inline client/flow_test.go:913]: we've had issues with 10 seconds here. what do you think about moving all 10s to a single constant?

Comment (sotnikov-s) [reply client/flow_test.go:913]: re this particular step and sleep duration: on reset node state, all operations taken from the history are processed almost immediately, bringing the nodes to up-to-date state in about a single second. the next operations are operations taken from get_operations and they are actually not needed -- those are duplicated operations of the ones just processed from the history, and they all fail/are skipped. so even the 10-seconds sleep here is an overkill, but it's suitable just to keep the logs in historical-ish order.

re all sleeps making constant: even though sleep durations are very different from each other, we need this difference for different steps take different time, e.g. dkg ceremony takes about 10-20 seconds, message signing takes 5-10 seconds, but in some cases (like in new tests I've made with selective get_operations) signing takes 2-3 seconds at most, etc. and making it constant of 20 seconds will dramatically increase the test time. let me take a look and maybe I can come up with an enum-like way of making the sleep durations constant

**Review by pr0n00gler — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Pull request merged**